### ISSUE TITLE
Pass URI to Credentials.getRequestMetadata

### ISSUE THREAD

**Issue by ejona86** [grpc/grpc-java#700]
`ServiceAccountJwtAccessCredentials` needs the URI, or requires the user to specify it ahead of time. Specifying it ahead of time is a pain and hard to know which strings are needed.

We will need to add a way to get the scheme and authority from `ClientCall` or `Channel`.

@anthmgoogle, FYI. @louiscryan and I agreed on this and have a gRPC-integration solution in mind. That means we will soon always be specifying the URI and passing the `defaultAudience` would no longer be necessary for our usage.

**Comment by ejona86** [comment]
This is necessary to implement [jwt_token_creds interop test](https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#jwt_token_creds)

**Comment by anthmgoogle** [comment]
So, if we don't need defaultAudience, I can remove that in a subsequent change prior to Beta. Agree that passing in on the call works best.

**Comment by nmittler** [comment]
@ejona86 I'm assuming that this will be run as an interceptor, correct?

**Comment by ejona86** [comment]
We will enhance to current `ClientAuthInterceptor`, but we need `Channel` support.

**Comment by nmittler** [comment]
@ejona86 Don't you have to pass the URL into the channel builder anyway? Why not just pass it to the `ClientAuthInterceptor` too? This seems like a one-off.

**Comment by nmittler** [comment]
Alternatively if we feel this is a more general need, it seems that this information could be made available via the `ClientCall` interface.  The `ClientAuthInterceptor` would just need to create the delegate call first, then it would have access to whatever it needs.

**Comment by ejona86** [comment]
Just passing it in to `ClientAuthInterceptor` is a losing battle. Once we have naming support (so things other than DNS) only the transport/naming SPI will know what the authority is. Even if they are only using DNS, there will be multiple ways to specify the hostname to connect to, like "dns://8.8.8.8/the-real-host.com".

When Louis and I discussed this, we originally thinking to add it to `ClientCall`. However, when reconnecting was implemented, it became obvious that had its own problems. A `ClientCall` hasn't chosen a transport before `start()` is called (and in the future, possibly even after `start()` is called). And really, it isn't the `ClientTransport` that knows the authority, but the `ClientTransportFactory`. If we retry a call on a different transport, we wouldn't want the `ClientCall` to suddenly start returning a different authority.

That leaves only load balancing. Would we ever need to know the path in order to know which authority we would use? I would say, "no," because even if we did, it would be broken because the same `ClientCall` could return different authorities over time.

And at that point `Channel` is really the only option left. But really it makes a lot of sense, because the `Channel` is a virtual connection to an endpoint; it only follows that it knows what the endpoint (authority) is.
